tests/tx/test_tx.py

```diff
@@ -94,6 +95,67 @@ def test_validation(self):
             self.assertTrue(block_from_chain.has_basic_block_parent())
         self.assertEqual(block_from_chain.get_next_block_best_chain(), None)
 
+    def test_checkpoint_validation(self):
+        from hathor.checkpoint import Checkpoint
+        from hathor.exception import InvalidNewTransaction
+        from hathor.transaction.transaction_metadata import ValidationState
+
+        # manually validate with sync_checkpoints=True
+        block1 = add_new_block(self.manager, propagate=False)
+        block1.validate_full(sync_checkpoints=True)
+        self.assertTrue(block1.get_metadata().validation.is_checkpoint())
+        self.manager.propagate_tx(block1)
+        del block1
+
+        # use on_new_tx with sync_checkpoints=True
+        block2 = self.manager.generate_mining_block()
+        block2.resolve()
+        self.assertTrue(self.manager.on_new_tx(block2, sync_checkpoints=True, partial=True, fails_silently=False))
+        self.assertEqual(block2.get_metadata().validation, ValidationState.CHECKPOINT_FULL)
+
+        # initialize a new manager and try to add a checkpoint block before its parents
+        block2_height = block2.get_metadata().height
+        manager2 = self.create_peer('testnet', checkpoints=[Checkpoint(block2_height, block2.hash)])
+        del block2._metadata
+        block2.storage = manager2.tx_storage
+        block2.set_height_hint(block2_height)
+        # it should fail if we don't pass sync_checkpoints=True
+        with self.assertRaises(InvalidNewTransaction):
+            manager2.on_new_tx(block2, partial=True, fails_silently=False)
+        # also test if it fails silently
+        self.assertFalse(manager2.on_new_tx(block2, partial=True, fails_silently=True))
+        # otherwise it should be accepted
+        self.assertTrue(manager2.on_new_tx(block2, sync_checkpoints=True, partial=True, fails_silently=False))
+        self.assertEqual(block2.get_metadata().validation, ValidationState.CHECKPOINT)
+
+        # otherwise it should fail if it's not in the checkpoints
+        manager3 = self.create_peer('testnet', checkpoints=[])
+        del block2._metadata
+        block2.storage = manager3.tx_storage
+        block2.set_height_hint(block2_height)
+        # failing silently should not raise an exception, but should return False
+        self.assertFalse(manager3.on_new_tx(block2, sync_checkpoints=True, partial=True, fails_silently=True))
+        # otherwise it should raise the correct exception
+        with self.assertRaises(InvalidNewTransaction):
+            manager3.on_new_tx(block2, sync_checkpoints=True, partial=True, fails_silently=False)
+
+    def test_set_height_hint(self):
```

Comment:
I feel it should be moved to the sync-v2 unittest.

Reply:
What do you mean by "sync-v2 unitest"? I kept it on this PR to keep the coverage.

Reply:
You moved to the next PR. Great!